### DISCUSSION TITLE
How is material composition specified in density correction files (line 3)?

### DISCUSSION THREAD

**Discussion by kstcmcjw** [nrc-cnrc/EGSnrc#882] [Questions] [answered]
Dear guiders, i got the concrete-brick density and the ratio of composition, and i will make a density file with ESTAR right now!
But what is the atomicNumber and massFraction mean? In the follow picture, i have rectangled with red color(Figure-1), the two data seem don't appeared in the download, the output-data(Figure-2), where are they? And what is the line 3 mean?

er, the output figure is a simulation, not a real ratio of composition!

![image](https://user-images.githubusercontent.com/96934217/175561346-7c36b5b8-a3c0-4e10-900a-c1dd11fa048a.png)
Figure-1

![image](https://user-images.githubusercontent.com/96934217/175562890-82983e6d-5ca9-4bcf-8615-11a71dafec78.png)
Figure-2

**Comment by ftessier** [comment, marked as answer]
Bear in mind that ratios (in ESTAR and in EGSnrc density correction file) are **mass** fractions. Line 3 list all atoms (by atomic number), and their respective mass fraction in the compound. So in the example of water we have atomic number `1` (hyrdogen), with a mass fraction of `0.111894`, and atomic number `8` (oxygen) with a mass fraction of `0.888106`.

For the example brick you have provided, you have to compute the total mass of the mixture and figure out the relative mass for each atomic number. This information is not provided by the ESTAR output from the web interface.

**Reply by ftessier** [reply]
For the record, and comparison with your own calculation, for your material mixture:
```
   Component    Fraction
               by Weight
           1    0.200000   CaO
           2    0.200000   Al2O3
           3    0.200000   CaCO3
           4    0.200000   SiO2
           5    0.200000   MgO
      Sum =     1.000000
```

I get the following atomic numbers and mass fractions (line 3):

```
  6 0.024001  8 0.433030 12 0.120607 13 0.105850 14 0.093487 20 0.223024
```

**Reply by kstcmcjw** [reply]
Sorry for reply so late, but i calculate the difference answer, is my atomic mass's problem?

The atomic mass is:
Ca=39.962591, Al=26.981538, Si=27.976927, Mg=23.985042, O=15.994915, C=12.000000
the Ca and Al all have 2 , and O has 10 , then the total atomic mass is： Ca×2+Al×2+O×10+Si+Mg+C=357.799377
then the ratio of C is 12/357.799377=0.03353835

It is different from your reslut, where is the mistake?

**Reply by kstcmcjw** [reply]
And there is another question puzzled me, that is the density file format, for example, should I select the four items? That is "Collision Stopping Power", "Radiative Stopping Power", "Total Stopping Power", and "Density Effect Parameter"
Should I select them all?

![image](https://user-images.githubusercontent.com/96934217/175805506-dba30d60-17f4-41ce-b53f-148bdd7ef3e7.png)

But i see the lead density file, it seems only one parameter.

![image](https://user-images.githubusercontent.com/96934217/175805658-b6e089ef-a78f-4e7b-9fed-889eb88af079.png)

**Reply by mxxo** [reply]
Only the density effect parameter is required (see point 6 here: https://github.com/nrc-cnrc/EGSnrc/wiki/Generating-density-effect-correction-files-with-ESTAR).

**Reply by kstcmcjw** [reply]
> 只需要单元格参数（参见这里的第 6 点：https://github.com/nrc-cnrc/EGSnrc/wiki/Generating-density-effect-correction-files-with-ESTAR）。

Thank you very much

**Reply by ftessier** [reply]
> It is different from your reslut, where is the mistake?

You have not considered the 0.2 mass fraction of _each_ compound?

**Reply by kstcmcjw** [reply]
I seem to understand that 20% is the proportion of the compound at the beginning. Now the proportion of the element is required to divide the weight of the compound by the molar mass to get the number of moles, and then add the same number of moles to get the total number of moles of a certain element. Then multiply by the molar mass and then divide by the mass of the total substance to get a single proportion of the total substance.

**Reply by kstcmcjw** [reply]
By the way, where is the 113 ?
![image](https://user-images.githubusercontent.com/96934217/176385351-f4e2d348-5b8f-4e3e-b64a-102436ed0476.png)

the submit file output is the Mean Excitation Energy, it is the seconed items.
![image](https://user-images.githubusercontent.com/96934217/176385475-08a78ab2-db4e-460f-b180-67e02b37d07c.png)

where is the  number Of Energies, it seems that not appeared in the output web.

**Reply by mxxo** [reply]
The output from ESTAR is not completely ready for use with EGSnrc, you have to add three lines at the start (including the number of energies), see point 7 here: https://github.com/nrc-cnrc/EGSnrc/wiki/Generating-density-effect-correction-files-with-ESTAR.

**Reply by kstcmcjw** [reply]
> The output from ESTAR is not completely ready for use with EGSnrc, you have to add three lines at the start (including the number of energies), see point 7 here: https://github.com/nrc-cnrc/EGSnrc/wiki/Generating-density-effect-correction-files-with-ESTAR.

Thank you, but I know the line 2 is  point 7, I mean i puzzled and i can't find the first item  "number Of Energies" , the ESTAR on-line's output does not contain the "number Of Energies", where is it?

**Reply by ftessier** [reply]
As @mxxo wrote, the ESTAR output is not formatted for direct use in EGSnrc. Either use the recommended list of 113 energies, or else count the number of energy values in your output. The [ESTAR instructions](https://github.com/nrc-cnrc/EGSnrc/wiki/Generating-density-effect-correction-files-with-ESTAR) are clear on this point:

> On the second line the `numberOfEnergies` is the number of energies that you saved into the file in step 4 (which should be 113)

**Reply by kstcmcjw** [reply]
> 要么使用推荐的 113 种能量列表，要么计算输出中能

er, I understand at last, the 113 is the kind of energy list which the ESTAR had defined, I am apologzie for my slow-witted, I should see the other density files in the $HEN_HOUSE\pegs4, and the other files are all 113.
Thanks to @mxxo and @ftessier, I find EGS has a magic and magnetic for me, have a nice day!

**Comment by kstcmcjw** [comment]
So, as the step of the caculate i have knew, i searched the concrete-brick structure, contains, ratio, searched the paper in the library on line, please allow me to discuess the concrete-brick's information.
The concrete-brick is consist of four to five compounds, that is about CaO( approximate 60%), SiO2(  approximate 20%), Fe2O3(7%-8%), Al2O3(10%-15%), and CO2(1%).
The raw materials for the manufacture of concrete bricks are cement, fine aggregate, coarse aggregate, admixture, admixture, and water. The cement is composed of five main components: CaO (60%), SiO2 (20%), Fe2O3 (7-8%), Al2O3 (10-15%), and CO2 (1%). But in the end, a concrete brick is formed by complex silicate hydration reaction of 100% main materials (cement, fine aggregate, coarse aggregate, admixture, admixture) and 10% water.
The density of a concrete brick is divided into three grades, the density of grade A is greater than 2.1g/cm^3, the density of grade B is 1.681~2.099g/cm^3, and the density of grade C is lower than 1.68g/cm^3.
Since the concrete bricks produced have strength density requirements, different strength density ratios are different. Take the concrete brick of 1.7g/cm^3 as an example, the final ratio is:
CaO(60-70%), SiO2(15-20%), Fe2O3(5-7%), Al2O3(7-12%), CO2(0.3%), MgO+Na2O+K2O(5%), and absent 10% water in this 100% ratio.

**Reply by ftessier** [reply]
In practice it is very difficult to define mixture materials (such as concrete, wood, plastics, some alloys, etc.) precisely in the absence of a certified composition from the manufacturer. Let alone for the specific brick you have in _your_ lab (short of performing your own chemical analysis!).

But keep in mind that to first order, what matters most is the bulk density, so focus first on getting that right: it is easy to measure as the mass over volume for a representative sample. Then perhaps choose proportions to err on the side of caution for the scenario at hand: for example, in a radiation protection calculation you might want to pick a composition that offer the _least_ shielding.

Finally, run simulations to **test** the sensitivity to the material composition. For example, run with and without water: does it make a difference for the quantity of interest (for the same bulk density)? I expect you will find that the relative proportions of the compounds of light elements have little impact for a fixed bulk density. That depends on the precision you are seeking of course: ultimately, the differences in composition will be detectable if you run simulation to a high enough degree of precision.

**Reply by kstcmcjw** [reply]
Thank you for your patient and meticulous reply!
These days I have consulted the chief technical director of the manufacturer, a nearly 60-year-old engineer who has worked hard for the cement and concrete business all his life. Yes, the specific element content of solid concrete bricks cannot be accurate to the percentage, most of which are an approximate concept, even if we determine the exact ratio of a certain concrete brick, it is also for a certain model, but a certain concrete brick can't to ensure the exact content ratio of the bricks in the production process, because the silicate hydration reaction is a relatively complex process, and solid concrete bricks are also divided into different types according to density and strength, so it is impossible to be very accurate. To calculate the content of elements, what can be done at present is only a rough approximation.
However, the density of solid concrete bricks can be detected by the detection mechanism. The density of the concrete bricks I use now is 1.79g/cm³, and a density file made by myself can only complete on this step.

Thank you very much for your reply! I wish you a good mood every day!
